cli/command/stack/deploy.go

```diff
@@ -164,6 +173,80 @@ func convertNetworks(
 	return nets
 }
 
+func convertVolumes(
+	serviceVolumes []string,
+	stackVolumes map[string]composetypes.VolumeConfig,
+	namespace string,
+) ([]mount.Mount, error) {
+	var mounts []mount.Mount
+
+	for _, volumeString := range serviceVolumes {
+		var (
+			source, target string
+			mountType      mount.Type
+			readOnly       bool
+			volumeOptions  *mount.VolumeOptions
+		)
+
+		// TODO: split Windows path mappings properly
+		parts := strings.SplitN(volumeString, ":", 3)
+
+		if len(parts) == 3 {
+			source = parts[0]
+			target = parts[1]
+			if parts[2] == "ro" {
+				readOnly = true
+			}
+		} else if len(parts) == 2 {
+			source = parts[0]
+			target = parts[1]
+		} else if len(parts) == 1 {
+			target = parts[0]
+		}
+
+		// TODO: catch Windows paths here
+		if strings.HasPrefix(source, "/") {
```

Comment:
😨 

Can we be a little more precise with this?
See https://github.com/docker/docker/pull/24182/files for an approach to this.
Basically copied windows and linux versions of `filepath.IsAbs` into something we can call from either platform.

As is this won't work on Windows.